rust/candid_parser/src/bindings/javascript.rs

```diff
@@ -238,39 +246,87 @@ fn pp_actor<'a>(ty: &'a Type, recs: &'a BTreeSet<&'a str>) -> RcDoc<'a> {
     }
 }
 
+pub fn pp_deprecation_comment<'a>() -> RcDoc<'a> {
+    str("/**").append(RcDoc::hardline())
+        .append(" * @deprecated Since `@dfinity/candid` v3.2.1, you can import IDL types directly from this module instead of using this factory function.")
```

Comment:
See https://js.icp.build/core/changelog#301---2025-07-22:
```suggestion
        .append(" * @deprecated Since `@dfinity/candid` v3.0.1, you can import IDL types directly from this module instead of using this factory function.")
```

Reply:
Unless you would like me to change the wording to keep `v3.0.1` and just explain that factory functions are no longer needed in general because of the changes in the version? So ignore being able to export from the module? It's just that this module is where these functions were used from, so without the `export` keywords I don't really feel like the factory functions are deprecated.

Reply:
The agent-js `v3.0.1` (aka [@dfinity/candid@3.0.1](https://www.npmjs.com/package/@dfinity/candid/v/3.0.1)) was the first version to have the fix for `instanceof`. This fix allows devs to import IDL types from wherever, including the JS bindings generated by `dfx`/`didc`, which are these bindings. The agent-js version is not linked to this repo.